File: Deck.java
import java.util.Collections;
import java.util.LinkedList;

public class Deck extends LinkedList<Card> {

    private int number;

    public Deck(int number) {
        this.number = number;
        reset();
    }

    public void reset() {
        clear();
        for (int d = 0; d < number; d++) {
            addDeck();
        }
        for (int c = 0; c < 7 + number; c++ /* haha aren't i funny */) {
            Collections.shuffle(this);
        }
    }

    private void addDeck() {
        for (int i = 2; i < 15; i++) {
            push(new Card(Card.Suit.CLUBS, i));
            push(new Card(Card.Suit.DIAMONDS, i));
            push(new Card(Card.Suit.HEARTS, i));
            push(new Card(Card.Suit.SPADES, i));
        }
    }
}

File: src/main/java/com/isaac/games/Deck.kt
package com.isaac.games

import java.util.Collections
import java.util.LinkedList

class Deck(private val number: Int) : LinkedList<Card>() {

    init {
        reset()
    }

    fun reset() {
        clear()
        for (d in 0 until number) {
            addDeck()
        }
        for (c in 0 until 7 + number) {
            Collections.shuffle(this)
        }
    }

    private fun addDeck() {
        for (i in 2..14) {
            push(Card(Card.Suit.CLUBS, i))
            push(Card(Card.Suit.DIAMONDS, i))
            push(Card(Card.Suit.HEARTS, i))
            push(Card(Card.Suit.SPADES, i))
        }
    }
}